docs/user/kind-example-config.yaml

```diff
@@ -1,23 +1,28 @@
 # this config file contains all config fields with comments
-apiVersion: kind.sigs.k8s.io/v1alpha1
 kind: Config
+apiVersion: kind.sigs.k8s.io/v1alpha2
+# 1 control plane node and 3 workers
+nodes:
+# the control plane node config
+- role: control-plane
+  # patch the generated kubeadm config with some extra settings
+  kubeadmConfigPatches:
+  - |
+    apiVersion: kubeadm.k8s.io/v1alpha3
+    kind: ClusterConfiguration
+    networking:
+      serviceSubnet: 10.0.0.0/16
+  # patch it further using a JSON 6902 patch
+  kubeadmConfigPatchesJson6902:
+  - group: kubeadm.k8s.io
+    version: v1alpha3
+    kind: ClusterConfiguration
+    patch: |
+      - op: add
+        path: /apiServerCertSANs/-
+        value: my-hostname
```

Comment:
Should we add some documentation for this? Explaining the syntax/use case?

Reply:
JSON 6902 is an IETF RFC we should link to that and probably kustomize somewhere in the docs, the GVK bit comes from k8s to define which object, and we're shimming kustomize to make this all inline.

FWIW, I have mixed feelings on this field. It will be a really handy and stable tool once the kubeadm API version remains stable I think 🤔